src/main/java/duchess/util/Storage.java

```diff
@@ -0,0 +1,112 @@
+package duchess.util;
+
+import duchess.tasks.Deadline;
+import duchess.tasks.Event;
+import duchess.tasks.Task;
+import duchess.tasks.ToDo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    private String filePath;
+
+    /**
+     * Create new Storage object.
+     *
+     * @param filePath file location for tasks.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Load tasks into ArrayList<Task>
+     * If no file identified, return an empty ArrayList<Task>.
+     *
+     * @return list Array of Task objects.
+     * @throws DuchessException if no file found and new file has to be created.
+     */
+    public ArrayList<Task> load() throws DuchessException {
+        ArrayList<Task> list = new ArrayList<>();
+        try {
+            File myObj = new File(filePath);
+            if (myObj.createNewFile()) {
+                System.out.println(myObj.getName() + " not found");
+                System.out.println("File created: " + myObj.getName() + "\n");
+                throw new DuchessException();
+            } else {
+                Scanner myReader = new Scanner(myObj);
+                while (myReader.hasNextLine()) {
+                    String data = myReader.nextLine();
+
+                    String[] input = data.split("");
+                    String type = input[1];
+                    String isDone = input[4];

```

Comment:
Would it be better if a different name was used, instead of naming the string as `isDone`
Perhaps using `markedStatus` or `doneChar` would be clearer?

Or maybe keeping the name but instead convert it to a boolean using `input[4].equals('X') ? true : false`

Please feel free to share your thoughts regarding my suggestions! 😊